include/seqan/parallel/enumerable_thread_local_iterator.h

```diff
@@ -0,0 +1,169 @@
+//     * Neither the name of Knut Reinert or the FU Berlin nor the names of
+//       its contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ==========================================================================
+// Author: Rene Rahn <rene.rahn@fu-berlin.de>
+// ==========================================================================
+
+#ifndef INCLDUE_SEQAN_ALIGN_PARALLEL_ENUMERABLE_THREAD_LOCAL_ITERATOR_H_
+#define INCLDUE_SEQAN_ALIGN_PARALLEL_ENUMERABLE_THREAD_LOCAL_ITERATOR_H_
+
+namespace seqan
+{
+
+// ============================================================================
+// Forwards
+// ============================================================================
+
+// ============================================================================
+// Tags, Classes, Enums
+// ============================================================================
+
+template <typename TEnumerableThreadLocal>
+class Iter<TEnumerableThreadLocal, EnumerableThreadLocalIterSpec>
+{
+public:
+
+    //-------------------------------------------------------------------------
+    // Member Types.
+
+    using TMap = typename TEnumerableThreadLocal::TMap;
+    using TMapIterator = typename IfC<std::is_const<TEnumerableThreadLocal>::value,
+                                      typename TMap::const_iterator,
+                                      typename TMap::iterator>::Type;
+
+    //-------------------------------------------------------------------------
+    // Member Variables.
+
+    TMapIterator _mInnerIter{};
+
+    //-------------------------------------------------------------------------
+    // Constructors.
+
+    Iter() = default;
+
+    Iter(typename TMap::iterator const & it) : _mInnerIter(it)
+    {}
+
+    Iter(typename TMap::const_iterator const & cit) : _mInnerIter(cit)
+    {}
+
+    Iter(TEnumerableThreadLocal & iterable) : Iter(iterable._map.begin())
+    {}
+
+    //-------------------------------------------------------------------------
+    // Member Functions.
+
+    inline auto*
+    operator->()
+    {
+        return &*(*this);
```

Comment:
![image](https://user-images.githubusercontent.com/2996863/34216353-ec75a84a-e59f-11e7-8039-bdab360f2c52.png)

Reply:
this works?

Reply:
below. It get's the element pointed to by the iterator member and returns an pointer to this element, so you can do the arrow semantic 👨‍🚀